### PR TITLE
#166731421 Feature pagination support

### PR DESCRIPTION
**Pull request by despeauxz**
#### What does this PR do?
This PR adds Pagination support to the entire application
#### Description of Task to be completed?
Check the `GET ratings route`
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#166731421
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by EMacco — DISMISSED** [review]

**Review by emmygozi — DISMISSED** [review]

**Review by EMacco — DISMISSED** [review]

**Review by EMacco — DISMISSED** [review]

**Review by emmygozi — DISMISSED** [review]

File: server/controllers/articles.js [hunk 1]
```diff
@@ -214,6 +219,58 @@ class ArticleController {
...
+      const { limit, offset } = await paginate.getQueryMetadata();
+
+      // Get the articleId
+      const article = await Article.findOne({ where: { slug } });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      const { id: articleId } = article.dataValues;
+
+      const ratings = await Rating.findAndCountAll({
```
Comment (fidelisojeah) [inline server/controllers/articles.js]: unfortunately, the `rating count` you are using here can't fly or make much sense.

if you want a count, you need to run a separate count query without the `limit` and `offset` but should include the `where clause`

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: server/controllers/articles.js (hunk 1 above)
Comment (despeauxz) [reply server/controllers/articles.js]: I'm on it boss

**Review by emmygozi — DISMISSED** [review]

File: server/controllers/articles.js [hunk 2]
```diff
@@ -214,6 +221,59 @@ class ArticleController {
...
+      const paginate = new Pagination(page, req.query.limit);
+      const { limit, offset } = await paginate.getQueryMetadata();
+
+      // Get the articleId
+      const article = await Article.findOne({ where: { slug } });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      const { id: articleId } = article.dataValues;
+      const ratingsDetails = await ratingsPayload(articleId);
```
Comment (fidelisojeah) [inline server/controllers/articles.js]: this is totally not necessary

you only need to count

`const count = Rating.count({where: {articleId}});`

and then get ratings
```javascript
const ratings = Rating.findAll({ 
  where: { articleId },
  limit,
  offset,
...
```

you should use same step for article

File: server/helpers/articlePayload.js [hunk 3]
```diff
@@ -1,12 +1,60 @@
...
+          as: 'profile',
+          attributes: ['firstname', 'lastname', 'bio', 'avatar']
+        }]
+      }
+    ],
+    group: ['Article.id', 'author.id', 'author->profile.id']
+  });
+};
+
+const findArticleCount = async () => Article.findAll({
```
Comment (fidelisojeah) [inline server/helpers/articlePayload.js]: unnecessary.

File: server/helpers/ratingPayload.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import models from '@models';
+
+const { Rating, User, Profile } = models;
+
+const findRatings = articleId => Rating.findAll({
```
Comment (fidelisojeah) [inline server/helpers/ratingPayload.js]: unnecessary

**Review by fidelisojeah — DISMISSED** [review]

File: server/controllers/articles.js (hunk 2 above)
Comment (despeauxz) [reply server/controllers/articles.js]: Thanks for the feedback, was really helpful

**Review by EMacco — DISMISSED** [review]

**Review by emmygozi — DISMISSED** [review]

**Review by emmygozi — DISMISSED** [review]

**Review by EMacco — DISMISSED** [review]

File: server/routes/api/articles.js [hunk 5]
```diff
@@ -20,4 +20,7 @@ articlesRouter.post('/', Token.authorize, ArticleController.create);
 articlesRouter.put('/:slug', Token.authorize, ArticleController.update);
 articlesRouter.delete('/:slug', Token.authorize, ArticleController.delete);
 
+articlesRouter.get('/:slug/rate', ArticleController.getArticleRatings);
+articlesRouter.post('/:slug/rate', trim, Token.authorize, ArticleController.rate);
```
Comment (fidelisojeah) [inline server/routes/api/articles.js:23]: duplicate
```suggestion

```

Comment (despeauxz) [reply server/routes/api/articles.js:23]: I'll fix it right now

**Review by adex001 — APPROVED** [review]
LGTM. Good work!

**Review by emmygozi — APPROVED** [review]

**Review by fidelisojeah — APPROVED** [review]

**Pull request merged**